set.mm

```diff
@@ -237805,6 +238091,472 @@ factorization into prime power factors (even if the exponents are
       ACDIUAJKLMNUUBUUC $.
   $}
 
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Simple groups
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+$)
+
+  $c SimpGrp $.
+
+  $( Extend class notation with the class of simple groups. $)
+  csimpg $a class SimpGrp $.
+
+  $( Define class of all simple groups.  A simple group is a group ( ~ df-grp )
+     with exactly two normal subgroups.  These are always the subgroup of all
+     elements and the subgroup containing only the identity ( ~ simpgnsgbid ).
+     (Contributed by Rohan Ridenour, 3-Aug-2023.) $)
+  df-simpg $a |- SimpGrp = { g e. Grp | ( NrmSGrp ` g ) ~~ 2o } $.
```

Comment:
Maybe I'm a bit pedantic here, but since this moves stuff to main, I'll have a go.

I think it is worth deciding what is our preferred way to write "there's exactly a pair".
Searching through `set.mm`, I see 44 occurrences of ` ~~ 2o`, but 119 occurrences of ``( # ` ... ) = 2``.
To me, the latter is more natural. Rohan chose the other way and there would be a handful of theorems to change if we were to decide to change.

Reply:
I used `~~ 2o` since that’s how ~df-prm does it.

Reply:
Good point, and it's important to keep the comparison with ~df-prm.
Also, `~~ 2o` is actually shorter than the other version, and it is more elementary.

But on the other hand, I believe that, compared to equinumerosity and ordinal numbers, the notion of the cardinality of a set, and the number 2 are better known and more accessible.

Anyway this is probably better discussed in a different forum. Maybe I'll open a generic issue for that instead.

Reply:
Yes, we should discuss this further in a separate Issue. I also had/have the need to handle sets with exactly 2 elements - therefore I defined a new symbol `PrPairs` for them in my mathbox.